charters/web5-working-group.md

```diff
@@ -0,0 +1,133 @@
+
+> Web5 brings decentralized identity and data storage to your applications. It lets devs focus on creating delightful user experiences, while returning ownership of data and identity to individuals.
+
+TBD has been building Web5 (nearly) since its inception. Web5 is the foundational layer of the TBD stack upon which decentralized applications, protocols, and services can be built. Today, Web5 is a collection of projects that are built on open standards pertaining to decentralized identity and self-sovereign technologies like Decentralized Identifiers (DIDs), and Verifiable Credentials (VCs), and Decentralized Web Nodes (DWNs) for secure personal storage and decentralized protocols.
+
+A key driver for this initiative is tbDEX, a use case that exemplifies the potential of the Web5 technology stack. tbDEX leverages Web5 to enable secure, decentralized value exchanges without intermediaries. This project highlights the practical applications and benefits of Web5 technologies, serving as a proof of concept that will guide the development and adoption of similar decentralized applications.
+
+Through this Working Group, we aim to continue building a robust and scalable framework for Web5, providing developers with the tools they need to build innovative and secure decentralized applications. This Working Group will coordinate with related working groups that may break off to focus on specific components of Web5, or have dependencies on this group's work.
+
+## Scope
+
+* Selection of standards pertaining to digital identifiers, credentials, storage, and associated protocols.
+
+* Implementation of a common set of standards using a well-defined API for the usage of these standards across programming languages.
+
+* Support for the needs of the TBD business unit, including, but not limited to the feature set required by [tbDEX](https://developer.tbd.website/projects/tbdex/).
+
+* Conformance infrastructure such as test vectors, test suites to encourage compatible implementations across programming languages.
+
+* Enable the development of Decentralized Applications (dApps) and Protocols using technologies such as Decentralized Identifiers (DIDs) and Verifiable Credentials (VCs).
+
+## Deliverables
+
+### 1. Web5 Specification
+
+**Description**: Develop and maintain the [Web5 Specification](https://github.com/TBD54566975/web5-spec/), that clearly specify:
+
+**DRI**: [Gabe Cohen](https://github.com/decentralgabe)
+
+**Scope of Work**:
+
+* Specification of the digital identity standards we will implement (e.g. specific DID methods, verifiable credential data models, credential issuance protocols, storage, etc.) along with specific implementation details (e.g. required features, minimal subsets, etc.).
+
+* An API used to interact with the digital identity standards we implement, able to be uptaken across Web5 SDKs.
+
+* Test vectors from which conformance to each feature in the specification can be achieved.
```

Comment:
>  An API used to...

If an API is part of the set of deliverables, then I would also recommend adding developer documentation and examples. These four things all go together:

1. API design
2. Test vectors
3. Doc comments
4. Example usage

Debatably, (3) & (4) are not a matter of conformance but of developer experience, but in practice we'll want to enable consistent developer experiences across implementations.

Reply:
BTW this is overlapped with Rust Core WG, which is not a problem, just calling it out

Reply:
Oh also, I would add clarity between the differences of "_An API_" and "_An API **design**_"

Reply:
After reading your comment in the proposed tbDEX Charter, @decentralgabe when you say "_An API_" is that an OpenAPI spec doc?

Reply:
yes - an OpenAPI spec doc. I will update this to make it clearer and call out the relation to the Rust WG.